tests/maths/maths.cpp

```diff
@@ -0,0 +1,249 @@
+    m.addComponent(c);
+    c->appendMathString(mathString);
+    std::string a = m.serialise(libcellml::FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Maths, modelWithTwoVariablesAndTwoInvalidMathStrings) {
+    const std::string e =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component\">"
+        "<variable name=\"variable1\"/>"
+        "<variable name=\"variable2\"/>"
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\"/>"
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\"/>"
+      "</component>"
+    "</model>";
+    std::string mathString =
+    "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">";
+    libcellml::Model m;
+    libcellml::ComponentPtr c = std::make_shared<libcellml::Component>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    c->setName("component");
+    v1->setName("variable1");
+    v2->setName("variable2");
+    c->addVariable(v1);
+    c->addVariable(v2);
+    c->appendMathString(mathString);
+    c->appendMathString(mathString);
+    m.addComponent(c);
+    std::string a = m.serialise(libcellml::FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+// 1.xiii.a
+TEST(Maths, modelWithTwoVariablesWithInitialValuesAndInvalidMathString) {
+    const std::string e =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component\">"
+        "<variable name=\"variable1\" initial_value=\"1.0\"/>"
+        "<variable name=\"variable2\" initial_value=\"-1.0\"/>"
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\"/>";
+      "</component>"
+    "</model>";
+    std::string mathString =
+    "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">";
+    libcellml::Model m;
+    libcellml::ComponentPtr c = std::make_shared<libcellml::Component>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    c->setName("component");
+    v1->setName("variable1");
+    v2->setName("variable2");
+    v1->setInitialValue("1.0");
+    v2->setInitialValue("-1.0");
+    c->addVariable(v1);
+    c->addVariable(v2);
+    c->appendMathString(mathString);
+    m.addComponent(c);
+    std::string a = m.serialise(libcellml::FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Maths, modelWithTwoVariablesWithInitialValuesAndValidMathString) {
+    const std::string e =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component\">"
+        "<variable name=\"A\" initial_value=\"1.0\"/>"
+        "<variable name=\"B\" initial_value=\"-1.0\"/>"
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+          "<apply>"
+             "<plus/>"
```

Comment:
Nothing wrong with this _per se_, but it might be better to do an equation like you'd get in a real model, e.g. C = A + B?